### PR TITLE
Use clientId and clientSecret as credentials to access token endpoint

### PR DESCRIPTION
**Pull request by robmckinnon**
For now only support Client Credentials Grant Type (OAuth 2.0): https://tools.ietf.org/html/rfc6749#section-4.4

Assume authentication using a client_id and client_secret: https://tools.ietf.org/html/rfc6749#section-2.3

Set client id and secret via env vars for now. Future implementation will store client_id, client_secret issued by ASPSP.

Set client id and secret for mock server in .env.sample.

Part of #CDRW-685

### CODE REVIEW

File: app/setup-account-request/obtain-access-token.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+};
+
+/*
+ * For now only support Client Credentials Grant Type (OAuth 2.0):
+ * https://tools.ietf.org/html/rfc6749#section-4.4
+ *
+ * Assume authentication using a client_id and client_secret:
+ * https://tools.ietf.org/html/rfc6749#section-2.3
+ */
+const postToken = async (authorisationServerHost, clientId, clientSecret) => {
```
Comment (fgyara2) [inline app/setup-account-request/obtain-access-token.js]: You should pass is scope[] as a parameter too

File: app/setup-account-request/obtain-access-token.js [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
   try {
     const response = await request
       .post(`${authorisationServerHost}/token`)
-      .set('authorization', 'xxx')
-      .type('form') // e.g. 'content-type': 'application/x-www-form-urlencoded'
+      .set('authorization', credentials(clientId, clientSecret))
+      .set('content-type', 'application/x-www-form-urlencoded')
       .send({
```
Comment (fgyara2) [inline app/setup-account-request/obtain-access-token.js]: You should send in scope and grant_type as parameters. grant_type should be client_credentials.

(See RFC 6749 Section 4.4.2)

File: app/setup-account-request/index.js [hunk 3]
```diff
@@ -4,9 +4,17 @@ const { postToken } = require('./obtain-access-token');
...
+  authorisationServerId ? {
+    clientId: process.env.ASPSP_AUTH_SERVER_CLIENT_ID,
+    clientSecret: process.env.ASPSP_AUTH_SERVER_CLIENT_SECRET,
+  } : null);
+
 const setupAccountRequest = async (authorisationServerId) => {
   const authorizationServerHost = await authorisationServerHost(authorisationServerId);
-  const response = await postToken(authorizationServerHost);
+  const { clientId, clientSecret } = await clientCredentials(authorisationServerId);
+  const response = await postToken(authorizationServerHost, clientId, clientSecret);
```
Comment (fgyara2) [inline app/setup-account-request/index.js]: Pass in 'accounts' to a scope parameter

File: app/setup-account-request/obtain-access-token.js (hunk 2 above)
Comment (robmckinnon) [reply app/setup-account-request/obtain-access-token.js]: Thanks, @fgyara2. As discussed scope and grant_type are set on the next two lines. :)

File: app/setup-account-request/obtain-access-token.js (hunk 1 above)
Comment (robmckinnon) [reply app/setup-account-request/obtain-access-token.js]: As discussed, we're hardcoding scope to 'accounts' within the function.

If needed for a future feature we can change to passing scope[] as parameter later.

**Pull request merged**